### PR TITLE
Add test cases for SECTION UNION defining multiple identical labels

### PR DESCRIPTION
**Pull request by Rangi42**
This is related to #1340, and is a blocker for PR #1341.

The easy part here was writing the tests. The hard part was fixing some Windows and macOS-specific issues with the test output:

- The error message for multiple identical exported labels, unusually, mention the object file names. Those are generated by `mktemp` in test.sh. On MinGW, they look like `"/tmp/foobar"` inside the bash shell, but that's just name mangling, and rgblink sees the unmangled `"C:/Users/RUNNER~1/ApData/Local/Temp/foobar"` path. So, I had to add an unmangle.cpp program (after trying alternatives like `echo` and `realpath`) so we can replace the arbitrary temp filename with `"section-union/the-test-case.o"` for comparison purposes.
- On macOS, I can't just do `sed -i 's|$objfile|the-test-case.o|g' "$outfile"`, because it uses BSD sed which demands a backup file prefix (or `''` for none). So we have to do `sed -i'' -e 's|$objfile|the-test-case.o|g' "$outfile"`.

In conclusion, I am sick of computers.

### CODE REVIEW

**Comment by ISSOtm** [comment]
What if the label is *exported*? That's when uniqueness is enforced.

That said, yep, the sorting is wrong... that should be fixed.

**Comment by Rangi42** [comment]
Oops, I forgot to double-colon them. The test case will need editing to allow for the .o output filenames.

**Comment by Rangi42** [comment]
Of course; I just haven't looked into what the sym sorting bug is yet.

**Comment by Rangi42** [comment]
> I think two test cases would be warranted here: one to check that duplicate labels are correctly sorted, and another to check that duplicate exported labels are rejected.

This is ready. :)
